src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/NFloat.cs

```diff
@@ -0,0 +1,82 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+
+#pragma warning disable SA1121 // We use our own aliases since they differ per platform
+#if TARGET_32BIT
+using NativeType = System.Single;
+#else
+using NativeType = System.Double;
+#endif
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// <see cref="NFloat"/> is an immutable value type that represents a floating type that has the same size
+    /// as the native integer size.
+    /// It is meant to be used as an exchange type at the managed/unmanaged boundary to accurately represent
+    /// in managed code unmanaged APIs that use a type alias for C or C++'s <c>float</c> on 32-bit platforms
+    /// or <c>double</c> on 64-bit platforms, such as the CGFloat type in libraries provided by Apple.
+    /// </summary>
+    [Intrinsic]
+    public readonly struct NFloat : IEquatable<NFloat>
+    {
+        private readonly NativeType _value;
+
+        /// <summary>
+        /// Constructs an instance from a 32-bit floating point value.
+        /// </summary>
+        /// <param name="value">The floating-point vaule.</param>
+        public NFloat(float value)
+        {
+            _value = value;
+        }
+
+        /// <summary>
+        /// Constructs an instance from a 64-bit floating point value.
+        /// </summary>
+        /// <param name="value">The floating-point vaule.</param>
+        public NFloat(double value)
+        {
+#if TARGET_32BIT
```

Comment:
The default behavior for floating point overflows is to convert to +Inf/-Inf. I think we should follow it here instead of throwing an exception.